src/game_main.cpp

```diff
@@ -50,36 +48,11 @@ namespace rigel {
 
 using namespace engine;
 using namespace sdl_utils;
+using namespace std::chrono;
 
 using RenderTargetBinder = renderer::RenderTargetTexture::Binder;
 
-namespace {
-
-template <typename Callback>
-class CallOnDestruction {
-public:
-  explicit CallOnDestruction(Callback&& callback)
-    : mCallback(std::forward<Callback>(callback))
-  {
-  }
-
-  ~CallOnDestruction() {
-    mCallback();
-  }
-
-  CallOnDestruction(const CallOnDestruction&) = delete;
-  CallOnDestruction& operator=(const CallOnDestruction&) = delete;
-
-private:
-  Callback mCallback;
-};
-
-
-template <typename Callback>
-[[nodiscard]] auto defer(Callback&& callback) {
-  return CallOnDestruction{std::forward<Callback>(callback)};
-}
-
+namespace misc{
```

Comment:
I decided to introduce a new `platform` namespace instead of `misc`, see https://github.com/lethal-guitar/RigelEngine/pull/561